conda/cli/main.py

```diff
@@ -51,7 +53,27 @@ def main_subshell(*args, post_parse_hook=None, **kwargs):
     parser = generate_parser(add_help=True)
     args = parser.parse_args(args, override_args=override_args, namespace=pre_args)
 
-    context.__init__(argparse_args=args)
+    # conda must determine if should be trying to read an input --file arg as part of
+    # setting up the context. To make this determination, check the conda sub command
+    # (`cmd`) that is getting executed. Conda should just do this for the following sub
+    # commands. Other commands may also support a --file argument, but these files are not
+    # environment spec files.
+    read_input_file = (
+        True if args.cmd in ("install", "create", "update") else False
+    )
+    # if we have a valid file argument, then we need to read it and pass its contents
+    # to the context. A --file might be:
+    # 1) a list of environments if coming from conda install/create/update
+    # 2) a single environment spec if coming from conda env
+    env_spec_paths = []
+    if read_input_file and getattr(args, "file", None):
+        if isinstance(args.file, str):
+            validate_file_exists(args.file)
```

Comment:
I don't think I like this. 

If the file does not exist, the plugin manager will return an error saying that no plugin was found. This causes an error for the test `tests/cli/test_env.py::test_conda_env_create_no_file ` since it expects a `EnvironmentFileNotFound` error instead. Maybe it's better for the test to be updated?